src/main/java/net/tiagofar78/prisonescape/game/PrisonEscapePlayer.java

```diff
@@ -97,7 +99,7 @@ public boolean hasIllegalItems() {
 //	########################################
 	
 	public boolean isWanted() {
-		return _isWanted;
+		return _isWanted || _inRestrictedArea;
```

Comment:
You can never do this. You have to call isInRestrictedArea() whenever isWanted() is called. The name of a method tells you what it is supposed to do. Don't change it's purpose.

Reply:
I feel like it would make sense to have a function in player that would say if he can be arrested or not rather than adding 2 conditions everytime

Reply:
so I thought that adding it in the isWanted made sense

Reply:
but maybe I can create another function in client that will do that @TiagoFar78  instead of adding it to that one

Reply:
Ok sure that's a nice solution.